### PR TITLE
Attempt to ensure that orchestrator logs make it out when terminating

### PR DESCRIPTION
**Pull request by andrisaar**
We know that currently sometimes when the container terminates not all logs make it outside to the launcher, hindering debugging.

This attacks this problem from three fronts:
  * Ensure that `oak-syslogd` starts _before_ the orchestrator and is terminated _after_ the orchestrator. Obviously if the syslogd gets killed before the orchestrator is, there is no hope of the logs making it out.
  * Force the orchestrator to flush logs via `journalctl --sync --flush` when it terminates. This should make any logs available to the syslogd.
  * Catch `SIGTERM` in the syslogd and terminate only after one loop of `sd_journal_wait()` has been empty; that is, there have been no new logs for one second.

It's a bit hacky, particularly the SIGTERM handling as (a) unix signals are inherently asynchronous so handling them is always a pain and (b) the systemd interface is synchronous and doesn't integrate well with rust async. The latter could be improved on, but I'm not sure it's worth (our) time.

Does it fix the issue of logs disappearing? Not sure. But it definitely won't make things worse.

### CODE REVIEW

**Review by jul-sh — APPROVED** [review]

**Comment by jul-sh** [comment]
looks like you may need to run format & clippy

**Comment by andrisaar** [comment]
> looks like you may need to run format & clippy

It's the bajillion changes required because `zerocopy` crate version changed and that's now percolating through _everything_ :(

File: oak_containers_syslogd/src/main.rs [hunk 1]
```diff
@@ -19,25 +19,51 @@
 mod log_relay;
 mod systemd_journal;
 
+use std::sync::Arc;
+
```
Comment (conradgrobler) [inline oak_containers_syslogd/src/main.rs:23]: super nit: for consistency avoid a space between different blocks of `use` statements.

**Review by conradgrobler — APPROVED** [review]
Thanks

**Pull request merged**